From 899267b63db039e483ca75c86f37fbbff69c0841 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:26:57 -0400
Subject: [tools] perf/evsel: Add missing overflow check for TRANSACTION

Message-id: <1408462094-14194-249-git-send-email-jolsa@redhat.com>
Patchwork-id: 88214
O-Subject: [PATCH RHEL7.1 BZ1131394 248/325] perf evsel: Add missing overflow check for TRANSACTION
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 87b955247d71975460774435241be3aa05218a7b
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Fri Nov 1 15:51:36 2013 +0200

description
===========
Add missing overflow check for PERF_SAMPLE_TRANSACTION in
perf_evsel__parse_sample().
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/evsel.c b/tools/perf/util/evsel.c
index 47bbf03..b121717 100644
--- a/tools/perf/util/evsel.c
+++ b/tools/perf/util/evsel.c
@@ -1481,6 +1481,7 @@ int perf_evsel__parse_sample(struct perf_evsel *evsel, union perf_event *event,
 
  data->transaction = 0;
  if (type & PERF_SAMPLE_TRANSACTION) {
+  OVERFLOW_CHECK_u64(array);
   data->transaction = *array;
   array++;
  }
-- 
1.7.1